app/Http/Controllers/ProspectActivityController.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Filters\ContactCustomerFilter;
+use App\Repositories\ContactCustomerRepository;
+use App\Repositories\ProspectActivityRepository;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+
+class ProspectActivityController extends Controller
+{
+    //
+    private $contactCustomerRepo;
+    private $prospectActivityRepo;
+    public function __construct(ContactCustomerRepository $contactCustomerRepository, ProspectActivityRepository $prospectActivityRepository)
+    {
+        $this->contactCustomerRepo = $contactCustomerRepository;
+        $this->prospectActivityRepo = $prospectActivityRepository;
+    }
+
+    public function index(ContactCustomerFilter $contactCustomerFilter)
+    {
+        // ->where('created_at', '=<', $date)
+        $prospects = $this->contactCustomerRepo->query($contactCustomerFilter)->whereHas('lastProspectActivity', function ($query) {
```

Comment:
I think this should be in ProspectActivityRepository and this filter works for only DSA's. U need to apply (isDsa, isDsaCaptain, isCoordinator) conditions